### PR TITLE
[KYUUBI #6266] Kyuubi BeeLine supports JSON output format.

### LINKED ISSUE #6266: [TASK][EASY] Kyuubi BeeLine supports JSON output format
### What's the level of this task?

EASY

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before creating

- [X] I have searched in the [task list](https://github.com/orgs/apache/projects/343) and found no similar tasks.

### Mentor

- [ ] I have sufficient expertise on this task, and I volunteer to be a mentor of this task to guide contributors through the task.

### Skill requirements

Knowledge of Java, Beeline, JSON

### Background and Goals

Similar to HIVE-20447, we'd like to have JSON output format support in Kyuubi BeeLine.

### Implementation steps

Refer to HIVE-20447

### Additional context

Introduction of [2024H1 Kyuubi Code Contribution Program](https://github.com/apache/kyuubi/issues/6232)

### PR DESCRIPTION
**Pull request by slfan1989**
# :mag: Description
## Issue References 🔗

This pull request fixes #6266

## Describe Your Solution 🔧

Kyuubi BeeLine supports JSON output format.

This pr cherrypick HIVE-20447, during this process, no incompatibility occurred in the code.

## Types of changes :bookmark:

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Test Plan 🧪

New UT

---

# Checklist 📝

- [x] This patch was not authored or co-authored using [Generative Tooling](https://www.apache.org/legal/generative-tooling.html)

**Be nice. Be informative.**

### CODE REVIEW

File: kyuubi-hive-beeline/src/test/java/org/apache/hive/beeline/TestJSONOutputFormat.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+  public ResultSet mockResultSet;
+  public TestBufferedRows.MockRow mockRow;
+
+  @Test
+  public final void testPrint() throws SQLException {
+    BeeLine mockBeeline = spy(BeeLine.class);
+    ArgumentCaptor<String> captureOutput = ArgumentCaptor.forClass(String.class);
+    Mockito.doNothing().when(mockBeeline).output(captureOutput.capture());
+    // when(mockBeeline.getOpts().getNumberFormat()).thenReturn("default");
+    // when(mockBeeline.getOpts().getConvertBinaryArrayToString()).thenReturn(true);
```
Comment (pan3793) [inline kyuubi-hive-beeline/src/test/java/org/apache/hive/beeline/TestJSONOutputFormat.java:52]: why comment?

Comment (slfan1989) [reply kyuubi-hive-beeline/src/test/java/org/apache/hive/beeline/TestJSONOutputFormat.java:52]: Thank you for reviewing the code! This is also the case in the original Hive code. I will remove this comment.

Comment (slfan1989) [reply kyuubi-hive-beeline/src/test/java/org/apache/hive/beeline/TestJSONOutputFormat.java:52]: https://github.com/apache/hive/pull/1169/files LINE-49

Comment (pan3793) [reply kyuubi-hive-beeline/src/test/java/org/apache/hive/beeline/TestJSONOutputFormat.java:52]: alright, leave it is fine then

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks, merged to master

**Pull request closed without merging**